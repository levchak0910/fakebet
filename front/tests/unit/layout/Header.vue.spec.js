import {createWrapper, disableFile, expectAsyncFunctionCalled} from "../__utils__";
import Component from "@/components/layout/Header";

disableFile();

describe("Header.vue", () => {

    let wrapper, authNav, notAuthNav, setEnLanguage, logout;
    let mountWrapper = (options = {}) => {
        wrapper = createWrapper(Component, {
            stubs: ["app-validating-input"],
            ...options,
        });
        authNav = wrapper.find("#auth");
        notAuthNav = wrapper.find("#not-auth");
        setEnLanguage = wrapper.find(".change-language:nth-of-type(1)");
        logout = wrapper.find("#logout");
    };

    describe("Testing snapshots", () => {

        it("User authorized", () => {
            mountWrapper({computed: {user: () => ({auth: true})}});

            expect(authNav.element).toMatchSnapshot();
        });

        it("User not authorized", () => {
            mountWrapper({computed: {user: () => ({auth: false})}});

            expect(notAuthNav.element).toMatchSnapshot();
        });

    });

    describe("Testing triggering methods", () => {

        it("setLanguage", () => {
            mountWrapper({methods: ["setLanguage"]});

            setEnLanguage.trigger("click");

            expect(wrapper.vm.setLanguage).toBeCalledWith("en");
        });

        it("userLogout", () => {
            mountWrapper({methods: ["userLogout"]});

            logout.trigger("click");

            expect(wrapper.vm.userLogout).toBeCalled();
        });

    });

    describe("Testing methods", () => {

        it("userLogout", async () => {
            mountWrapper({methods: ["_userLogout"]});

            wrapper.vm.userLogout();

            expect(wrapper.vm._userLogout).toBeCalled();
            expectAsyncFunctionCalled(wrapper, "$router.replace", "matches");
        });

        it("setLanguage", () => {
            mountWrapper();

            wrapper.vm.setLanguage("en");

            expect(document.documentElement.lang).toBe("en");
            expect(localStorage.getItem("language")).toBe("en");
            expect(wrapper.vm.$i18n.locale).toBe("en");
        });

    });

    describe("Testing hooks", () => {

    });

});