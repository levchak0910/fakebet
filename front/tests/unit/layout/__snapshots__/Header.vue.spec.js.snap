// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header.vue Testing snapshots User authorized 1`] = `
<nav
  id="auth"
>
  <p>
    <router-link-stub
      to="/about"
    >
      <a>
        About
      </a>
    </router-link-stub>
  </p>
  <p>
    Language:
    <span
      class="change-language"
    >
      English
    </span>
    <span>
      /
    </span>
    <span
      class="change-language"
    >
      Russian
    </span>
  </p>
  <p>
    <router-link-stub
      to="/bets"
    >
      <a>
        Bets
      </a>
    </router-link-stub>
  </p>
  <p>
    <span
      id="logout"
    >
      Logout
    </span>
  </p>
</nav>
`;

exports[`Header.vue Testing snapshots User not authorized 1`] = `
<nav
  id="not-auth"
>
  <p>
    <router-link-stub
      to="/about"
    >
      <a>
        About
      </a>
    </router-link-stub>
  </p>
  <p>
    Language:
    <span
      class="change-language"
    >
      English
    </span>
    <span>
      /
    </span>
    <span
      class="change-language"
    >
      Russian
    </span>
  </p>
  <p>
    <router-link-stub
      to="/auth"
    >
      <a>
        Authentication
      </a>
    </router-link-stub>
  </p>
</nav>
`;
